src/playlist-merge.js

```diff
@@ -0,0 +1,583 @@
+
+      if (!newPlaylist) {
+        removedMediaGroupPlaylists.push({
+          type,
+          group,
+          label,
+          playlist: oldPlaylist
+        });
+      }
+    });
+  });
+
+  return removedMediaGroupPlaylists;
+};
+
+/**
+ * Returns any new playlists that do not account for all timelines (where all timelines is
+ * the max number of timelines seen across all of the passed in playlists).
+ *
+ * @param {Array} playlists the playlists to look through
+ *
+ * @return {Array} any playlists that do not account for all timelines
+ */
+export const getIncompletePlaylists = (playlists) => {
+  const timelines = {};
+  const playlistToTimelines = {};
+
+  playlists.forEach((playlist) => {
+    if (!playlist.segments) {
+      return;
+    }
+
+    const playlistTimelines = playlist.segments.reduce((acc, segment) => {
+      acc[segment.timeline] = true;
+      timelines[segment.timeline] = true;
+
+      return acc;
+    }, {});
+
+    // All playlists should have a unique ID sourced from the required attribute
+    // Representation@id from the MPD.
+    playlistToTimelines[playlist.attributes.NAME] = playlistTimelines;
+  }, {});
+
+  const numNewTimelines = Object.keys(timelines).length;
+
+  if (numNewTimelines === 0) {
+    return [];
+  }
+
+  return playlists.filter((playlist) =>
+    Object.keys(playlistToTimelines[playlist.attributes.NAME]).length < numNewTimelines);
+};
+
+/**
+ * Gets a flattened array of media group playlists.
+ *
+ * @param {Object} manifest the main manifest object
+ *
+ * @return {Array} the media group playlists
+ */
+export const getMediaGroupPlaylists = (manifest) => {
+  let mediaGroupPlaylists = [];
+
+  forEachMediaGroup(manifest, SUPPORTED_MEDIA_TYPES, (properties, type, group, label) => {
+    mediaGroupPlaylists = mediaGroupPlaylists.concat(properties.playlists || []);
+  });
+
+  return mediaGroupPlaylists;
+};
+
+/**
+ * @typedef {Object} MediaGroupPlaylistIdentificationObject
+ * @property {string} type - the media group type
+ * @property {string} group - the media group...group
+ * @property {string} label - the media group label
+ * @property {Object} playlist - the playlist within the group
+ */
+
+/**
+ * Given a list of playlists and a manifest object, returns an array of objects with all
+ * of the necessary media group properties to identify it within a manifest.
+ *
+ * @param {Object} config
+ * @param {Array} config.playlists the playlist objects
+ * @param {Object} config.manifest the main manifest object
+ *
+ * @return {MediaGroupPlaylistIdentificationObject[]}
+ *         media group playlist identification objects
+ */
+export const getMediaGroupPlaylistIdentificationObjects = ({ playlists, manifest }) => {
+  if (!playlists.length) {
+    return [];
+  }
+
+  const idObjects = [];
+
+  forEachMediaGroup(manifest, SUPPORTED_MEDIA_TYPES, (properties, type, group, label) => {
+    properties.playlists.forEach((playlist) => {
+      if (playlists.includes(playlist)) {
```

Comment:
`.includes` isn't available on IE11 either, unfortunately.

Reply:
I usually use `.some` in place of `.includes`